test/unit/util/application-logging.test.js

```diff
@@ -58,27 +54,26 @@ tap.test('truncate', (t) => {
     { value: [], type: 'array' },
     { value: function () {}, type: 'function' }
   ]
-  negativeTests.forEach(({ value, type }) => {
-    t.test(`should not truncate ${type}`, (t) => {
-      const newValue = loggingUtils.truncate(value)
-      t.same(value, newValue)
-      t.end()
+  await Promise.all(
+    negativeTests.map(async ({ value, type }) => {
+      await t.test(`should not truncate ${type}`, () => {
+        const newValue = loggingUtils.truncate(value)
+        assert.deepEqual(value, newValue)
+      })
     })
-  })
+  )
```

Comment:
Since there aren't any before and after test hooks in the parent test, I imagine this works well. But if we did have the hooks, I'm not certain this approach would work. I don't think the hooks would get applied correctly through the `.map`.

I have been unrolling these sorts of patterns, e.g.:

```js
for (const negativeTest of negativeTests) {
	const { value, type } = negativeTest
	await t.test(type, () => assert.equal(value, 'whatever'))
}
```

Reply:
yea i can do that. I wasn't sure